crates/openvino/src/core.rs

```diff
@@ -260,4 +218,53 @@ mod core_tests {
         let model = core.read_model_from_buffer(model, None);
         assert!(model.is_ok());
     }
+
+    #[test]
+    fn test_get_core_properties_supported() {
+        let core = Core::new().unwrap();
+        let supported_keys = vec![
+            SupportedProperties,
+            AvailableDevices,
+            OptimalNumberOfInferRequests,
+            RangeForAsyncInferRequests,
+            RangeForStreams,
+            DeviceFullName,
+            DeviceCapabilities,
+        ];
+        for key in supported_keys {
+            let supported_properties = core.get_property(&DeviceType::CPU, &key);
+            assert!(
+                supported_properties.is_ok(),
+                "Failed on supported key: {:?}",
+                &key
+            );
+        }
+    }
+
+    #[test]
+    fn test_get_core_properties_rw() {
+        let core = Core::new().unwrap();
+        let rw_keys = vec![
+            CacheDir,
+            NumStreams,
+            Affinity,
+            InferenceNumThreads,
+            HintEnableCpuPinning,
+            HintEnableHyperThreading,
+            HintPerformanceMode,
+            HintSchedulingCoreType,
+            HintInferencePrecision,
+            HintNumRequests,
+            EnableProfiling,
+            HintExecutionMode,
+        ];
+        for key in rw_keys {
+            let supported_properties = core.get_property(&DeviceType::CPU, &PropertyKey::Rw(&key));
```

Comment:
```suggestion
            let supported_properties = core.get_property(&DeviceType::CPU, &key.into()));
```

This should be possible if we add a `From<RwPropertyKey>` (?).

Reply:
Added